File: backend/app/models/operations.py
"""只读运维能力视图；不包含密钥值或可执行配置。"""

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


class ValidationBackendInfo(BaseModel):
    model_config = ConfigDict(extra="forbid")

    name: str
    display_name: str
    available: bool
    supported_languages: list[str] = Field(default_factory=list)
    supported_profiles: list[str] = Field(default_factory=list)
    executes_untrusted_code: bool
    requires_user_configuration: bool
    unavailable_reason: str | None = None
    health_status: Literal["healthy", "degraded", "unavailable"]
    last_health_check_at: datetime
    safety_boundary: str
    documentation_url: str
    registered_runner_count: int | None = None


class ValidationProfilesResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    backend: str
    profiles: list[str]


class VersionDiagnostics(BaseModel):
    config: str
    prompt: str
    rule: str
    tool_schema: str
    review_policy: str
    patch_policy: str


class SystemDiagnostics(BaseModel):
    model_config = ConfigDict(extra="forbid")

    version: str
    provider: str
    default_model: str
    database_schema_current: bool
    worker_status: Literal["running", "idle"]
    artifact_directory_writable: bool
    langsmith_enabled: bool
    security_mode: Literal["restricted", "unsafe_local"]
    patch_max_files: int
    patch_max_changed_lines: int
    retention_days: int
    validation_backends: list[ValidationBackendInfo]
    configured_secrets: dict[str, bool]
    versions: VersionDiagnostics

File: frontend/src/types/operations.ts
import type { ReviewTask } from "./review";

export type AppPage = "dashboard" | "history" | "validation" | "settings";

export interface ReviewTaskListResponse {
  items: ReviewTask[];
  total: number;
  page: number;
  page_size: number;
}

export type BackendHealthStatus = "healthy" | "degraded" | "unavailable";

export interface ValidationBackendInfo {
  name: string;
  display_name: string;
  available: boolean;
  supported_languages: string[];
  supported_profiles: string[];
  executes_untrusted_code: boolean;
  requires_user_configuration: boolean;
  unavailable_reason?: string | null;
  health_status: BackendHealthStatus;
  last_health_check_at: string;
  safety_boundary: string;
  documentation_url: string;
  registered_runner_count?: number | null;
}

export interface VersionDiagnostics {
  config: string;
  prompt: string;
  rule: string;
  tool_schema: string;
  review_policy: string;
  patch_policy: string;
}

export interface SystemDiagnostics {
  version: string;
  provider: string;
  default_model: string;
  database_schema_current: boolean;
  worker_status: "running" | "idle";
  artifact_directory_writable: boolean;
  langsmith_enabled: boolean;
  security_mode: "restricted" | "unsafe_local";
  patch_max_files: number;
  patch_max_changed_lines: number;
  retention_days: number;
  validation_backends: ValidationBackendInfo[];
  configured_secrets: Record<string, boolean>;
  versions: VersionDiagnostics;
}

export interface ProviderModelInfo {
  id: string;
  owned_by?: string | null;
}

export interface ModelCatalogResponse {
  provider: string;
  default_model: string;
  models: ProviderModelInfo[];
}
